packages/webapp/src/components/Form/NumberInput/NumberInput.tsx

```diff
@@ -0,0 +1,253 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import { ChangeEvent, ReactNode, useLayoutEffect, useMemo, useRef, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import InputBase, { type InputBaseProps } from '../InputBase';
+import { MdKeyboardArrowDown, MdKeyboardArrowUp } from 'react-icons/md';
+import styles from './styles.module.scss';
+import { clamp, countDecimalPlaces } from './utils';
+
+export type NumberInputProps = {
+  /**
+   * The initial value of the input
+   */
+  value?: number | string;
+  /**
+   * function called on change with value represented as a number as the paramter.
+   */
+  onChange?: (valueAsNumber: number) => void;
+  /**
+   * function called when input is blurred
+   */
+  onBlur?: () => void;
+  /**
+   * Controls grouping of numbers over 1000 with the thousands separator.
+   */
+  useGrouping?: boolean;
+  /**
+   * Controls whether or not a decimal is allowed as input. If set to false, users can only enter whole numbers.
+   */
+  allowDecimal?: boolean;
+  /**
+   * The locale to use for number formatting.
+   */
+  locale?: string;
+  /**
+   * Number of decimal places to round on blur. Ignored if used with a step value
+   */
+  roundToDecimalPlaces?: number;
+  /**
+   * The unit to display on right side of input
+   */
+  unit?: ReactNode;
+  /**
+   * Amount to step up or down. Renders a stepper on right side with up and down buttons when step is greater than 0.
+   * If allowDecimal is false, then step is rounded down to the nearest whole number.
+   */
+  step?: number;
+  /**
+   * Max value of input. Only applicable if used with step
+   */
+  max?: number;
+  /**
+   * Min value of input. Only applicable if used with step
+   */
+  min?: number;
```

Comment:
The purpose of max and min here is to disable the up/down buttons to prevent the user from going past the min or max. So it doesn't make sense to use without step. But for validation purposes we can still pass min or max to react hook form, like: 

```
<NumberInputRHF 
   name='some-name'
   rules={   
      max: {
        value: 20,
        message: 'error message'
    },    
     min: {
       value: 5,
       message: 'error message'
    }
/>
```